### PR TITLE
Rainbow-100: use Z80SIO instead of Z80DART; make UCSD Pascal boot again

### PR DESCRIPTION
**Pull request by Bavarese**
Incorporate change from Z80DART to Z80SIO proposed by Edstrom. Also make UCSD Pascal boot again.

### CODE REVIEW

File: src/mame/drivers/rainbow.cpp [hunk 1]
```diff
@@ -1405,26 +1408,26 @@ WRITE_LINE_MEMBER(rainbow_state::mpsc_irq)
 WRITE8_MEMBER(rainbow_state::comm_bitrate_w)
 {
 	m_dbrg_A->str_w(data & 0x0f);  // PDF is wrong, low nibble is RECEIVE clock (verified in SETUP).
-	printf("\nRECEIVE bitrate = %02x HEX\n",data & 0x0f);
+	printf("\n(COMM.) receive bitrate = %d ($%02x)\n", comm_rates[data & 0x0f] , data & 0x0f);
```
Comment (cuavas) [inline src/mame/drivers/rainbow.cpp]: Could you please replace these `printf` statements with one of the MAME-provided things?  You have a choice of:
* Configurable logging from `logmacro.h` (preferred)
* Direct calls to `logerror`
* `osd_printf_verbose` etc. (least preferred as it doesn't automatically provide device context and isn't configurable)

**Comment by Bavarese** [comment]
Changed printf to logerror as desired (actually a leftover from debugging). Haven't used the configurable option so far, but will look into it.

File: src/mame/drivers/rainbow.cpp [hunk 2]
```diff
@@ -2415,91 +2420,73 @@ WRITE8_MEMBER(rainbow_state::z80_diskcontrol_w)
...
-//      printf("%i <- SELECTED DRIVE...\n", m_unit);
 	}
 
 	if (selected_drive == INVALID_DRIVE)
 	{
-		printf("(m_unit = %i)   ** SELECTED DRIVE ** INVALID. (selected drive = %i)\n", m_unit, selected_drive);
-		m_unit = INVALID_DRIVE;
-		m_floppy = nullptr;
-	}
+		printf("(m_present_drive = %i)   ** SELECTED DRIVE ** INVALID. (selected drive = %i)\n", m_present_drive, selected_drive);
```
Comment (cuavas) [inline src/mame/drivers/rainbow.cpp]: This one still snuck through.

**Comment by Bavarese** [comment]
Done.

File: src/mame/drivers/rainbow.cpp (hunk 2 above)
Comment (Bavarese) [inline src/mame/drivers/rainbow.cpp]: Done

**Pull request merged**